src/utils/redux-helpers/action-creator-helpers.js

```diff
@@ -0,0 +1,345 @@
+    }
+
+    assert(_.isUndefined(resultPropTypes));
+    assert(_.isUndefined(requestPropTransform));
+    assert(_.isUndefined(resultPropTransform ));
+
+    return makeBasicActionCreator({ type, propTypes: requestPropTypes });
+};
+
+const makeBasicActionCreator = ({ type, propTypes }) => {
+
+    assert(isSimpleActionType(type));
+    assert(_.isUndefined(propTypes) || _.isObject(propTypes));
+
+    const actionCreator = (rawPayload) => {
+
+        assert(_.isUndefined(rawPayload) || _.isObject(rawPayload));
+
+        const payload = makePayload({ rawPayload, propTypes });
+
+        return makeAction({ type, payload });
+    };
+
+    actionCreator.TYPE = type;
+
+    return actionCreator;
+};
+const makeExtendedActionCreator = ({ type, propTypes, propTransform, after }) => {
+
+    assert(isSimpleActionType(type));
+    assert(_.isUndefined(propTypes) || _.isObject(propTypes));
+    assert(_.isUndefined(propTransform) || _.isFunction(propTransform));
+    assert(_.isUndefined(after) || isAfter(after));
+
+    const actionCreator = (rawPayload) => {
+
+        assert(_.isUndefined(rawPayload) || _.isObject(rawPayload));
+
+        return (dispatch, getState) => {
+
+            const payload = makePayload({ rawPayload, propTypes, propTransform, getState });
+
+            const dispatchResult = dispatch(makeAction({ type, payload }));
+
+            return processAfter({
+
+                after, type, payload, dispatch, getState
+
+            }).then(_.constant(dispatchResult));
+        };
+    };
+
+    actionCreator.TYPE = type;
+
+    return actionCreator;
+};
+const makeAsyncActionCreator = (
+    { type, requestPropTypes, resultPropTypes, requestPropTransform, resultPropTransform , async, after }) => {
+
+    assert(isAsyncActionType(type));
+    assert(_.isUndefined(requestPropTypes) || _.isObject(requestPropTypes));
+    assert(_.isUndefined(resultPropTypes) || _.isObject(resultPropTypes));
+    assert(_.isUndefined(requestPropTransform) || _.isFunction(requestPropTransform));
+    assert(_.isUndefined(resultPropTransform ) || _.isFunction(resultPropTransform ));
+    assert(_.isFunction(async));
+    assert(_.isUndefined(after) || isAfter(after));
+
+    const actionCreator = (rawRequest) => {
+
+        assert(_.isUndefined(rawRequest) || _.isObject(rawRequest));
+
+        return (dispatch, getState) => {
+
+            const request = makePayload({
+                rawPayload: rawRequest,
+                propTypes: requestPropTypes,
+                propTransform: requestPropTransform,
+                getState
+            });
+
+            dispatch(actionCreator.request(request));
+
+            return Promise.resolve(
+
+                async(request, { dispatch, getState })
+
+            ).then(
+                (rawResult) => {
+
+                    // Rethrow if Error
+                    if (rawResult instanceof Error) {
+                        throw rawResult;
+                    }
```

Comment:
Added these lines from the original source because wrapping the `async(...)` in `Promise.resolve()` seems to not cause an error to go to the second handler (line 150)

Reply:
This has been reverted -- I'm not sure what was going wrong when I wrote this -- it may be something with the hot module reloader